HW8/Holovach/password validation.py

```diff
@@ -0,0 +1,32 @@
+import re
+
+def password_validation():
+
+    while 1:
+
+        password = input("Enter the password:")
+
+        if len(password) < 6:
+            print("Your password's length should be more than 6 characters!")
+            continue
+        elif len(password) > 16:
+            print("Your password's length should be less than 16 characters!")
+            continue
+        elif bool(re.findall("[a-z]", password)) != True:
+            print("Your password has to contain a sign from this range[a-z]!")
+            continue
+        elif bool(re.findall("[A-Z]", password)) != True:
+            print("Your password has to contain a sign from this range[A-Z]!")
+            continue
+        elif bool(re.findall("[0-9]", password)) != True:
+            print("Your password has to contain a number from this range[0-9]!")
+            continue
+        elif bool(re.findall("[$,#,@]", password)) != True:
+            print("Your password has to contain a sign from this range[$,#,@]!")
+            continue
+        else:
+            return "You are welcome!"
+    
+    
+             
+print(password_validation())

```

Comment:
import re

def validate_password(input_password):
    
    flag = True
    
    while flag:  
        if len(input_password)<6 or len(input_password)>16:
            break
        elif not re.search("[a-z]",input_password):
            break
        elif not re.search("[0-9]",input_password):
            break
        elif not re.search("[A-Z]",input_password):
            break
        elif not re.search("[$#@]",input_password):
            break
        # Returns a match where the string contains a white space character
        elif re.search("\s",input_password):  
            break
        else:
            return f"{input_password} is valid password"
            flag = False
            break

    if flag:
        return f"{input_password} is invalid password"


print(validate_password("ER45jknkjbyhdfy"))
print(validate_password("ER45@hjkyutt TYTIU566"))
print(validate_password("DFG567#oijl89"))
##########################3

import re

password = input("Enter your password:\n")

if re.match("^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[$#@])(?=.{6,16}$).*", password):
    print("Nice pass! Psss, I won't tell anyone ;)")
else:
    print("Password is not valid. Check if you have:"
          """
    At least 1 letter between[a-z] and 1 letter between[A-Z].
    At least 1 number between[0-9].
    At least 1 character from [$#@].
    Minimum length 6 characters.
    Maximum length 16 characters.
          """
           )